test/Microsoft.TemplateEngine.Edge.UnitTests/LocalizationTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge.Settings;
+using Microsoft.TemplateEngine.Mocks;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects;
+using Microsoft.TemplateEngine.TestHelper;
+using Microsoft.TemplateEngine.Utils;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Edge.UnitTests
+{
+    public class LocalizationTests
+    {
+        [Theory]
+        [InlineData(null, "name")]
+        [InlineData("de-DE", "name_de-DE:äÄßöÖüÜ")]
+        [InlineData("tr-TR", "name_tr-TR:çÇğĞıIİöÖşŞüÜ")]
+        public void TestLocalizedTemplateName(string locale, string expectedName)
+        {
+            _ = LoadHostWithLocalizationTemplates(out SettingsLoader settingsLoader);
+
+            IReadOnlyList<TemplateInfo> localizedTemplates = settingsLoader.UserTemplateCache.GetTemplatesForLocale(locale, null);
+
+            Assert.True(localizedTemplates.Count != 0);
```

Comment:
it would be good to specify the message in asserts. It helps when debugging failed tests.

Reply:
Added where possible. Most of the assert calls don't have an overload that takes a message though.